File: rock_paper_scissors_backend/src/match_model.ts
import PlayerMatchModel from "./player_match_model";

class MatchModel {
  id: string;
  players: PlayerMatchModel[];
  roundsWinners: string[];
  winner: string;

  constructor(id: string) {
    this.id = id;
    this.players = [];
    this.roundsWinners = [];
    this.winner = '';
  }
}

export default MatchModel;

File: rock_paper_scissors_frontend/lib/models/match_model.dart
import 'models.dart';

class MatchModel {
  final String id;
  final List<PlayerMatchModel> players;
  final List<String> roundsWinners;
  final String winner;

  MatchModel({
    required this.id,
    this.players = const [],
    this.roundsWinners = const [],
    this.winner = '',
  });

  factory MatchModel.fromJson(Map<String, dynamic> json) {
    return MatchModel(
      id: json['id'],
      players: json['players'] != null
          ? (json['players'] as List).map((json) => PlayerMatchModel.fromJson(json)).toList()
          : [],
      roundsWinners: json['roundsWinners'] != null ? List<String>.from(json['roundsWinners']) : [],
      winner: json['winner'],
    );
  }
}
